### ISSUE TITLE
Add possibility  to delete docs from Elasticsearch.

### ISSUE THREAD

**Issue by jann** [Jeffail/benthos#373]
We recently had the use case to write and delete documents from elasticsearch using benthos. Currently I dont see the option in benthos to be able to delete documents from elasticsearch. Do you think this would be a useful feature? I am thinking of contributing this feature to this open source project, but would like to ask before if  you have thoughts about it. Thank you.

**Comment by Jeffail** [comment]
Hey @jann, this would be great. We could add an elasticsearch `cache` type that would enable this.

**Comment by jann** [comment]
Hi @Jeffail. Would it be possible to dynamically write or delete documents based on a value in the metadata? What are your thoughts about this? Or would you rather cover this use case by running two benthos in parallel, one writing into elasticsearch the other deleting from it and filtering the corresponding delete/write messages before in a processor.

**Comment by DpoBoceka** [comment]
As an alternative, you're able to have `http` processor to perform a crafted REST API request to your target Elasticsearch. 
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete.html

**Comment by DpoBoceka** [comment]
Shouldn't be much of a problem given `http` processor supports interpolation in it.

**Comment by Jeffail** [comment]
> Would it be possible to dynamically write or delete documents based on a value in the metadata? 

Yeah, if it's a cache type then we can do all kinds of weird and freaky things with it.

**Comment by twmb** [comment]
Closing — this was addressed in #436, which added support for delete (and other operations) in the elasticsearch output.
